src/logger.ts

```diff
@@ -176,14 +177,29 @@ class Logger {
       let logFunction = this.printer[functionName];
       let allMsgs = '';
       let allStyles: string[] = [];
+
+      /** Flush all prefix and styles into msgsAndStyles
+       * Note: there may not be styles associated with a message or prefix!
+       */
       for (let [msg, style] of this.prefixesAndStyles) {
-        allMsgs += `%c[${msg}]`;
-        allStyles.push(style);
+        if (style) {
+          allMsgs += `%c[${msg}]`;
+          allStyles.push(style); // Only add style to allStyles if present
+        } else {
+          allMsgs += `[${msg}]`;
+        }
+      }
+      if (allMsgs.length > 0) {
+        allMsgs += '%c '; // space between prefixes and rest of logged message
+        allStyles.push('color: inherit; background-color: transparent;');
```

Comment:
maybe we can break this out into a `const` in your `./styles.ts` (and import it into this file) in case we need to format other whitespace things in the future.